File: loadercomplex-forge-18/src/main/java/com/enderzombi102/loadercomplex/forge18/impl/entity/ForgeItemEntity.java
package com.enderzombi102.loadercomplex.forge18.impl.entity;

import com.enderzombi102.loadercomplex.api.entity.ItemEntity;
import com.enderzombi102.loadercomplex.api.item.ItemStack;
import com.enderzombi102.loadercomplex.forge18.impl.item.ForgeItemStack;

public class ForgeItemEntity extends ForgeEntity implements ItemEntity {
	private final net.minecraft.entity.ItemEntity backingEntity;

	public ForgeItemEntity(net.minecraft.entity.ItemEntity backingEntity) {
		super( backingEntity );
		this.backingEntity = backingEntity;
	}

	@Override
	public ItemStack getStack() {
		return new ForgeItemStack( this.backingEntity.getStack() );
	}

	@Override
	public void setStack(ItemStack stack) {
		this.backingEntity.setStack( (net.minecraft.item.ItemStack) stack.getStack() );
	}
}

File: loadercomplex-forge/src/ver182/kotlin/com/enderzombi102/loadercomplex/forge182/impl/entity/ForgeItemEntity.kt
package com.enderzombi102.loadercomplex.forge182.impl.entity

import com.enderzombi102.loadercomplex.api.minecraft.item.ItemStack
import com.enderzombi102.loadercomplex.forge182.impl.item.ForgeItemStack
import net.minecraft.entity.ItemEntity

class ForgeItemEntity(private val backingEntity: ItemEntity?) : ForgeEntity(
	backingEntity!!
),
	com.enderzombi102.loadercomplex.api.minecraft.entity.ItemEntity {
	override fun getStack(): ItemStack {
		return ForgeItemStack(backingEntity!!.stack)
	}

	override fun setStack(stack: ItemStack) {
		backingEntity!!.stack = stack.stack as net.minecraft.item.ItemStack
	}
}
